### ISSUE TITLE
Temporary directory and recursive removing

### ISSUE THREAD

**Issue by matteo-cristino** [zen-fs/core#99]
Hi there, I'm new to zenFS and I'm testing it for the moment with node, but found out some interesting stuff that may be also of your interest.

This is the code I'm using
```js
import { promises as fs } from "@zenfs/core";
import path from "path-browserify";

// if this is line commented out then fs.mkdtemp will return error:
// ErrnoError: ENOENT: No such file or directory, '/tmp'
// should this be created directly by fs.mkdtemp if /tmp is not found?
await fs.mkdir("/tmp");

const tmpdir = await fs.mkdtemp("random-folder-");
await fs.mkdir(path.join(tmpdir, "foo"));
await fs.writeFile(path.join(tmpdir, "foo", "test.txt"), "hello world!");

// This fails with
// ErrnoError: ENOTEMPTY: Directory is not empty, '/tmp/random-folder-1724408696549-5pjlwuyb46e/foo'
// seems that recursive options is not forwarded in the recursrive calls
await fs.rm(tmpdir, { recursive: true });
```
with `@zenfs/core@0.16.3`.

As written in the comment:
* `fs.mkdtemp` fails if I firstly do not create the `/tmp` directory, maybe this can be done directly in that function if it does not exists?
* `fs.rm` recursive options seems not to works with neseted folders and files

**Comment by mho22** [comment]
I encountered the same recursivity problem with a different code. I also tried to implement a method to make this recursivity work : 

```javascript
function removeDirectoryRecursively( directory )
{
    if( fs.existsSync( directory ) )
    {
        let files = fs.readdirSync( directory, { withFileTypes : true } );

        files.forEach( file =>
        {
            const path = `${directory}/${file.path}`;

            const stat = fs.statSync( path );

            if( stat.isFile() )
            {
                fs.rmSync( path );
            }

            if( stat.isDirectory() )
            {
                removeDirectoryRecursively( path );
            }
        } );

        files = fs.readdirSync( directory, { withFileTypes : true } );

        if( files.length == 0 )
        {
            fs.rmdirSync( directory );
        }
    }
}
```

But even this is returning `Directory is not empty` for every directories to remove.

**Comment by matteo-cristino** [comment]
Opened PR #100 that should address the recursively removal

**Comment by james-pre** [comment]
Fixed in #100. Thanks a lot!
